app/dawn-hub/Views/OnboardingViews/Step2_BlockSettingsView/BlockSettingsView.jsx

```diff
@@ -39,57 +44,56 @@ class BlockSettingsView extends Component {
 
 	componentDidMount() {
 		const { setupLifecycle: { blockSetupSeen } } = this.props;
+
 		if (blockSetupSeen) {
 			const {
-				antiSuite: {
-					enable_anti_tracking,
-					enable_ad_block,
-					enable_smart_block,
-				},
+				enable_anti_tracking,
+				enable_ad_block,
+				enable_smart_block,
 				blockingPolicy
 			} = this.props;
 
-			const decodedPolicy = this.decodeBlockingPolicy(blockingPolicy);
-
 			this.setState({
-				kindsOfTrackers: decodedPolicy,
+				kindsOfTrackers: blockingPolicy,
 				enable_ad_block,
 				enable_anti_tracking,
 				enable_smart_block
 			});
 		}
 	}
 
-	decodeBlockingPolicy = (blockingPolicy) => {
+	componentDidUpdate() {
+		const {
+			recommendedChoices,
+			enable_anti_tracking,
+			enable_ad_block,
+			enable_smart_block,
+			kindsOfTrackers
+		} = this.state;
+
+		if (!recommendedChoices && kindsOfTrackers === BLOCKING_POLICY_RECOMMENDED && enable_ad_block === true && enable_anti_tracking === true && enable_smart_block === true) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({ recommendedChoices: true });
+		} else if (recommendedChoices && (kindsOfTrackers !== BLOCKING_POLICY_RECOMMENDED || enable_ad_block !== true || enable_anti_tracking !== true || enable_smart_block !== true)) {
+			// eslint-disable-next-line react/no-did-update-set-state
```

Comment:
It would be good if we can avoid needing to disable this linting rule. Updating state in `componentDidUpdate` will trigger another render right after one has been completed, which according to the linter docs justifying the rule may lead to layout thrashing.

`recommendedChoices` can be derived from the other bits of state anyway, so it would be cleaner to dispense with it. when we only store state values that can't be derived from other state values, and derive others from these as needed, we eliminate the risk of a derived piece of state being incorrectly out of sync due to forgetting to update it or a bug in the update logic.

my suggestion:
* get rid of `recommendedChoices`
* replace it with a `_recommendedChoices()` method that tells you whether the current combo of selections matches the recommended choices and use that method to determine whether the RC checkbox should be checked in `render()`
* get rid of `componentDidUpdate()` - with this setup, you don't need it anymore. whenever a user changes a setting, the corresponding bits of state gets updated, `render()` runs, and `_recommendedChoices()` spits out the appropriate checkbox value

Reply:
Thank you for this comment, definitely better this way! I've updated this to using function `recommendedChoicesActive()` for determining whether the checkbox should be checked.

Reply:
Great, thanks!